### PR TITLE
sno AgentClusterInstall and InfraEnvironment updates

### PR DESCRIPTION
**Pull request by jnunyez**
- Generalized AgentClusterInstall and patched  using patchesstrategicMerge.
-  Added in overlay AgentclusterInstall example of configuring dual IPv4/IPv6 networking stack 
- Added dual stack example for NTP sources in patched Infraenv

### CODE REVIEW

File: overlay/sno/patches/aci.yaml [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+metadata:
+  name: cluster
+  namespace: cluster-name
+spec:
+  clusterDeploymentRef:
+    name: rna4
+  imageSetRef:
+    name: openshift-v4.10.16
+  # dual stack network config
+  networking:
```
Comment (novacain1) [inline overlay/sno/patches/aci.yaml:13]: The other possible alternative here is single stack IPv6.  Meaning more specifically, no IPv4.

File: overlay/sno/patches/infraenv-patch.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@
 - op: replace
   path: /spec/baseDomain
   value: 'cloud.lab.eng.bos.redhat.com'
+# IPv4/IPv6 NTP
+- op: replace
+  path: /spec/additionalNTPSources
+  value: ['10.18.100.10', 'fd00:6:6:11::52']
```
Comment (novacain1) [inline overlay/sno/patches/infraenv-patch.yaml:10]: Similar comments as with the AgentClusterInstall CRD.

**Review by novacain1 — COMMENTED** [review]
Added comments for Single stack IPv6.

File: overlay/sno/patches/infraenv-patch.yaml (hunk 2 above)
Comment (jnunyez) [reply overlay/sno/patches/infraenv-patch.yaml:10]: Add detail about this for user leveraging our example:

```suggestion
# IPv4/IPv6 dual stack (configured) and (IPv4,IPv6)-only stacks supported
- op: replace
  path: /spec/additionalNTPSources
# Comment/delete next line for (IPv4,IPv6)-only stacks 
  value: ['10.18.100.10', 'fd00:6:6:11::52']
# Uncomment next line for (IPv4)-only stacks 
# value: ['10.18.100.10']
# Uncomment next line for (IPv6)-only stacks 
# value: ['fd00:6:6:11::52']
```

File: overlay/sno/patches/infraenv-patch.yaml [hunk 3]
```diff
@@ -4,6 +4,10 @@
 - op: replace
   path: /spec/baseDomain
   value: 'cloud.lab.eng.bos.redhat.com'
+# IPv4/IPv6 NTP
```
Comment (jnunyez) [inline overlay/sno/patches/infraenv-patch.yaml:7]:
```suggestion
# IPv4/IPv6 dual, IPv4-only, IPv6-only supported
```

File: overlay/sno/patches/aci.yaml [hunk 4]
```diff
@@ -1,10 +1,26 @@
...
+kind: AgentClusterInstall
+metadata:
+  name: cluster
+  namespace: cluster-name
+spec:
+  clusterDeploymentRef:
+    name: rna4
+  imageSetRef:
+    name: openshift-v4.10.16
+  # dual stack network config
```
Comment (jnunyez) [inline overlay/sno/patches/aci.yaml:12]:
```suggestion
  # IPv4/IPv6 dual stack and (IPv4,IPv6)-only stacks supported
```

File: overlay/sno/patches/infraenv-patch.yaml (hunk 2 above)
Comment (novacain1) [reply overlay/sno/patches/infraenv-patch.yaml:10]: Looks good.

File: overlay/sno/patches/aci.yaml (hunk 4 above)
Comment (novacain1) [inline overlay/sno/patches/aci.yaml:12]: Looks good.

File: overlay/sno/patches/aci.yaml (hunk 1 above)
Comment (jnunyez) [reply overlay/sno/patches/aci.yaml:13]: Tried to add detailed about this for user leveraging this example:

```suggestion
  # IPv4/IPv6 dual stack (default) and (IPv4,IPv6)-only stacks supported
  networking:
    clusterNetwork:
  # delete/comment next two lines for IPv6-only stack
      - cidr: 10.128.0.0/14
        hostPrefix: 23
  # delete/comment next two lines for  IPv4-only stack
      - cidr: fd01::/48
        hostPrefix: 64
    serviceNetwork:
  # delete/comment next line for IPv6-only stack
      - 172.30.0.0/16
  # delete/comment next line for a IPv4-only stack
      - fd02::/112
    machineNetwork:
  # delete/comment next line for IPv6-only stack
      - cidr: 172.17.0.0/24
  # delete/comment next line for IPv4-only stack
      - cidr: fd00:6:6:2060::0/64
  # yamllint disable-line rule:line-length
  sshPublicKey: 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDzcf7gr7S7jVWKJr1/NsLvTBR/JNmqpWwygq/v4rP+QObIZME5EmYNYw4jwGk/anDxFff3XcPVKYjNarv7xULn3F+yOmZxZbmFrXTZC+iWArGD56W79CTaCxPsZies32/BitoLZwqPZf6lsBKaMB0LOdlrKfZFP7x6e/PR9or+1+1PWHfP3QJmJNfLjMzFxrUM/UBkMR8knL8dFsAkvOTfdRm/X56enMlmqzqN5kWhSYwhj7YCwssUNMF+qWVYaggKyRbxFcSv3ytit4Ygw4HRPfSNiyBHxFv2EmSmXCiawv1JidHVIrfFjqHRh/dtBlMAF3Bwd7pYoQ8ELvpIoRXZ02pFab1pmNO/KXFn7F8RW2wZYmTa+YLrXPQrC087R50Mto/cWhwaRuXqtnAgMfO78hxo8lwqbDwfcbSOr8D9mYM5FVJ+cH4LDDwFzuICFh3kp82ALF44ZzOpcsXVY4P4wPzXqda4lQjwx4sN6/ZDZcFnrad48+PXp/wEWTMfPbM= core@metal3'
```

**Comment by jnunyez** [comment]
> Added comments for Single stack IPv6.

Added comments for the user for the different options (IPv4, IPv6)-only and dual stacks for the sake of clarity.

**Review by natifridman — APPROVED** [review]
LGTM

File: overlay/sno/patches/aci.yaml (hunk 1 above)
Comment (novacain1) [reply overlay/sno/patches/aci.yaml:13]: Looks OK

**Review by novacain1 — APPROVED** [review]

**Pull request merged**